design/cluster-update.md

```diff
@@ -0,0 +1,209 @@
+Adding a new OSD is fairly simple since the OSD pod logic already supports it:
+* If the entire node is being added, ensure the node is added to the CRUSH map: `ceph osd crush add-bucket {bucket-name} {type}`
+* For each OSD:
+  * Register, format, add OSD to the crush map and start the OSD process like normal
+  * Wait for all placement groups to reach `active+clean` state, meaning data migration is complete.
+
+#### Removing Storage
+Removing storage is a more involved process and it will also be done one OSD at a time to ensure the cluster returns to a clean state.
+Of special note for removing storage is that a check should be performed to ensure that the cluster has enough remaining storage to recover (backfill) the entire set of objects from the OSD that is being removed.
+If the cluster does not have enough space for this (e.g., it would hit the `full` ratio), then the removal should not proceed.
+
+For each OSD to remove, the following steps should be performed:
+* mark the OSD as `out` with `ceph osd out <osd.id>`, which will trigger data migration from the OSD.
+* wait for all data to finish migrating from the OSD, meaning all placement groups return to the `active+clean` state
+* stop the OSD process and remove it from monitoring
+* remove the OSD from the CRUSH map: `ceph osd crush remove <osd.id>`
+* delete the OSD's auth info: `ceph auth del <osd.id>`
+* delete the OSD from the cluster: `ceph osd rm <osd.id>`
+* delete the OSD directory from local storage (if using `dataDirHostPath`): `rm -fr /var/lib/rook/<osd.id>`
+
+If the entire node is being removed, ensure that the host node is also removed from the CRUSH map:
+```console
+ceph osd crush rm <host-bucket-name>
+```
+
+#### Completion
+After all storage updates are completed, both additions and removals, then we can once again enable scrubbing:
+```console
+ceph osd unset noscrub
+ceph osd unset nodeep-scrub
+```
+
+### Placement Groups
+The number of placement groups in the cluster compared to the number of OSDs is a difficult trade-off without knowing the user's intent for future cluster growth.
+The general rule of thumb is that you want around 100 PGs per OSD.
+With less than that, you have potentially unbalanced distribution of data with certain OSDs storing more than others.
+With more PGs than that, you have increased overhead in the cluster because more OSDs need to coordinate with each other, impacting performance and reliability.
+
+It's important to note that **shrinking** placement group count (merging) is still **not supported** in Ceph.
+Therefore, you can only increase the number of placement groups (splitting) over time.
+
+If the cluster grows such that we have too few placement groups per OSD, then we can consider increasing the number of PGs in the cluster by incrementing the `pg_num` and `pgp_num` for each storage pool.
```

Comment:
I think this should be postponed until we either have a) a logic to calculate pgs or b) allow the user to configure it (as it is not that simple to calculate with a "changing" data workload to find good values).

Reply:
Agreed, I don't see much being done with PG counts until #560 is tackled.